app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -43,35 +43,25 @@ class SchoolDisciplineDashboard extends React.Component {
     });
   }
 
-  studentDisciplineIncidentCounts(incidentCategory) {
+  studentDisciplineIncidentCounts(incidents) {
     let studentDisciplineIncidentCounts = {};
-    const selectedChartData = this.getChartData(this.state.selectedChart);
-    const incidents = incidentCategory ?
-                      selectedChartData.disciplineIncidents[incidentCategory] :
-                      this.filterIncidentDates(this.props.schoolDisciplineEvents);
+
+    //if a user selects a category and then moves to a time range with no incidents within that category, return an empty object
+    if (!incidents) return studentDisciplineIncidentCounts;
     incidents.forEach((incident) => {
       studentDisciplineIncidentCounts[incident.student_id] = studentDisciplineIncidentCounts[incident.student_id] || 0;
       studentDisciplineIncidentCounts[incident.student_id]++;
     });
     return studentDisciplineIncidentCounts;
   }
 
-  getChartData(selectedChart) {
-    //This chart data is filtered based on the date selection and passed to the
-    //student table and chart renders below
-    const filteredEvents = this.filterIncidentDates(this.props.schoolDisciplineEvents);
-    return {
-      type: selectedChart,
-      disciplineIncidents: _.groupBy(filteredEvents, selectedChart),
-      title: "Incidents by " + selectedChart};
-  }
-
-  sortChartKeys(chartKeys) {
```

Comment:
This was a kludge and no longer necessary now that we have more clarity about how we want users to interact with the dashboard